server/conn.go

```diff
@@ -1144,6 +1144,42 @@ func (cc *clientConn) handleLoadStats(ctx context.Context, loadStatsInfo *execut
 	return loadStatsInfo.Update(prevData)
 }
 
+// handleIndexAdvise does the index advise work and returns the advise result for index.
+func (cc *clientConn) handleIndexAdvise(ctx context.Context, indexAdviseInfo *executor.IndexAdviseInfo) error {
+	if cc.capability&mysql.ClientLocalFiles == 0 {
+		return errNotAllowedCommand
+	}
+	if indexAdviseInfo == nil {
+		return errors.New("Index Advise: info is empty")
+	}
+
+	err := cc.writeReq(indexAdviseInfo.Path)
+	if err != nil {
+		return err
+	}
+	var prevData, curData []byte
+	for {
+		curData, err = cc.readPacket()
+		if err != nil && terror.ErrorNotEqual(err, io.EOF) {
+			return err
+		}
+		if len(curData) == 0 {
+			break
+		}
+		prevData = append(prevData, curData...)
+	}
+	if len(prevData) == 0 {
+		return errors.New("Index Advise: infile is empty")
+	}
```

Comment:
Can we extract the the upper parts as a function. It is almost the same with `handleLoadStats`.

Reply:
Yep, i will do it now.